build: update to husky@6

damingerdai opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Commit:
build: update to husky@6

Upgrade husky from 5.0.1 to 6.0.0

gkalpak reviewed (approved):
Thx, @damingerdai!
Should we switch to `prepare` npm script (instead of `postinstall`) as is the new recommended way (after typicode/husky#890)?

gkalpak commented:
BTW, you need to add a commit message body (of at least 20 characters) to pass linting.

damingerdai commented:
> Should we switch to `prepare` npm script (instead of `postinstall`) as is the new recommended way (after [typicode/husky#890](https://github.com/typicode/husky/pull/890))?

I agree with you, but I am not sure whether I should complete the changes in this pr.

damingerdai commented:
@gkalpak please review

gkalpak commented:
@damingerdai, I don't see any changes. Did you forget to push them maybe?

damingerdai commented:
> @damingerdai, I don't see any changes. Did you forget to push them maybe?

i just add commit message body.

gkalpak commented:
Oh, I see, sorry 😅
I thought you implemented switching to `prepare` and was looking for code changes 😇

I think it is reasonable to switch to `prepare` as part of this PR, since it is a new recommendation in husky 6, so it makes sense to make the change along with the updating imo.

Commit:
build: install husky in `prepare` script instead of `postinstall`

With typicode/husky#890, the recommended way to install husky is in the
`prepare` script instead of the `postinstall`. This commit moves
the husky installation to the `prepare` script to align with the new
recommendation.

damingerdai commented:
@gkalpak please review

gkalpak commented on package.json:
```diff
@@ -36,7 +36,8 @@
     "ng-dev": "node dev-infra/ng-dev",
-    "ng-dev:dev": "ts-node --transpile-only -- dev-infra/cli.ts"
+    "ng-dev:dev": "ts-node --transpile-only -- dev-infra/cli.ts",
+    "prepare": "husky install"
```
Please move this script just below `postinstall` (since that is conceptually closer to how the scripts are called).

gkalpak reviewed (changes requested):
Thx for making the changes, @damingerdai :+1:

Can you please change the second commit message to mention typicode/husky#890, where the recommendation to switch to `prepare` is made. For example:

> build: install husky in \`prepare\` script instead of \`postinstall\`
>
> With typicode/husky#890, the recommended way to install husky is in the
> \`prepare\` script instead of the \`postinstall\`. This commit moves
> the husky installation to the \`prepare\` script to align with the new
> recommendation.

damingerdai commented:
@gkalpak thanks for your suggestions. please review again.

gkalpak reviewed (approved):
Thx again, @damingerdai :+1:

LGTM, except you need to rebase on latest master to make CI happy :smiley:

damingerdai commented:
@gkalpak i have rebased on latest master but there still are two ci failed

gkalpak reviewed (approved):
Thx again, @damingerdai!
The latest CI failures were flakes. I have restarted the job and it passes now 🚀
